components/com_tags/src/Service/Router.php

```diff
@@ -89,47 +89,67 @@ public function preprocess($query)
         }
 
         // Get query language
-        $language = isset($query['lang']) ? $query['lang'] : '*';
+        $lang = isset($query['lang']) ? $query['lang'] : '*';
 
         // Set the language to the current one when multilang is enabled and item is tagged to ALL
-        if (Multilanguage::isEnabled() && $language === '*') {
-            $language = $this->app->get('language');
+        if (Multilanguage::isEnabled() && $lang === '*') {
+            $lang = $this->app->get('language');
         }
 
-        if (isset($query['view']) && $query['view'] == 'tags') {
-            if (isset($query['parent_id']) && isset($this->lookup[$language]['tags'][$query['parent_id']])) {
-                $query['Itemid'] = $this->lookup[$language]['tags'][$query['parent_id']];
-            } elseif (isset($this->lookup[$language]['tags'][0])) {
-                $query['Itemid'] = $this->lookup[$language]['tags'][0];
-            }
-        } elseif (isset($query['view']) && $query['view'] == 'tag') {
-            if (isset($query['id'])) {
-                if (!is_array($query['id'])) {
-                    $query['id'] = [$query['id']];
-                }
-
-                $id = ArrayHelper::toInteger($query['id']);
-                sort($id);
-
-                if (isset($this->lookup[$language]['tag'][implode(',', $id)])) {
-                    $query['Itemid'] = $this->lookup[$language]['tag'][implode(',', $id)];
+        foreach ([$lang, '*'] as $language) {
```

Comment:
```suggestion
        foreach (array_unique([$lang, '*']) as $language) {
```
If multilanguage is not enabled, `$lang`will be `'*'`, so the `foreach` will loop over `['*', '*']`. We can avoid this with using array_unique.